presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriteUtils.java

```diff
@@ -423,7 +423,15 @@ private static void checkWritable(
 
     public static Path getTableDefaultLocation(HdfsContext context, SemiTransactionalHiveMetastore metastore, HdfsEnvironment hdfsEnvironment, String schemaName, String tableName)
     {
-        Optional<String> location = getDatabase(metastore, schemaName).getLocation();
+        Database database = metastore.getDatabase(schemaName)
+                .orElseThrow(() -> new SchemaNotFoundException(schemaName));
+
+        return getTableDefaultLocation(database, context, hdfsEnvironment, schemaName, tableName);
+    }
+
+    public static Path getTableDefaultLocation(Database database, HdfsContext context, HdfsEnvironment hdfsEnvironment, String schemaName, String tableName)
```

Comment:
Should the commit message be "Add variant of HiveWriteUtils.getTableDefaultLocation"?